### PR TITLE
usage-data: maintain state of progress when scraping events

### PR DESCRIPTION
**Pull request by coury-clark**
Closes https://github.com/sourcegraph/sourcegraph/issues/39089

Capture state of scraping jobs progress. Adds a new table `event_logs_scrape_state` that will record the bookmark of the highest event_id that was successfully sent. If no state is found a new state row will be initialized at the current event. Older events will not be backfilled.

## Test plan

To test locally:

Set up gcp credentials 
```
gcloud auth application-default login
```
Start `sg` 
```
GOOGLE_APPLICATION_CREDENTIALS="$HOME/.config/gcloud/application_default_credentials.json" sg start ...
```

Add configuration to the site config
```
    "exportUsageTelemetry": {
      "enabled": true,
      "topicProjectName": "sourcegraph-dogfood",
      "topicName": "usage-data-testing"
    },
```

Log lines will indicate job progress
```
[         worker] INFO worker.export-usage-telemetry telemetry/telemetry_job.go:127 fetching events from bookmark {"bookmark_id": 21333}
[         worker] INFO worker.export-usage-telemetry telemetry/telemetry_job.go:138 telemetryHandler executed {"event count": 5, "maxId": 21338}
```

Check bookmark table
```
select bookmark_id from event_logs_scrape_state order by id limit 1;
```

### CODE REVIEW

File: enterprise/cmd/worker/internal/telemetry/telemetry_job_test.go [hunk 1]
```diff
@@ -222,6 +230,74 @@ func TestHandlerLoadsEvents(t *testing.T) {
...
+
+	t.Run("first execution of handler should return first event", func(t *testing.T) {
+		handler.sendEventsCallback = expectEvents(t, want[:0])
+
+		err = handler.Handle(ctx)
+		if err != nil {
+			t.Fatal(err)
+		}
+	})
+	t.Run("second execution of handler should return second event", func(t *testing.T) {
```
Comment (coury-clark) [inline enterprise/cmd/worker/internal/telemetry/telemetry_job_test.go:283]: There is a problem with these tests - I'm working on it.

File: enterprise/cmd/worker/internal/telemetry/telemetry_job.go [hunk 2]
```diff
@@ -279,3 +297,35 @@ func getInitialAdminEmail(ctx context.Context, store database.UserEmailsStore) (
...
+
+func (s *bmStore) GetBookmark(ctx context.Context) (_ int, err error) {
+	tx, err := s.Transact(ctx)
+	if err != nil {
+		return 0, err
+	}
+	defer func() { err = tx.Done(err) }()
+
+	val, found, err := basestore.ScanFirstInt(tx.Query(ctx, sqlf.Sprintf("select bookmark_id from event_logs_scrape_state order by id limit 1;")))
+	if !found {
```
Comment (chwarwick) [inline enterprise/cmd/worker/internal/telemetry/telemetry_job.go:322]: Feels like there could be an edge case here where a transient error causes the select to fail and it ends up in inserting a new new row and skipping some events.

**Review by chwarwick — APPROVED** [review]
Looks good, but I do think you need an additional error check in GetBookmark.

File: enterprise/cmd/worker/internal/telemetry/telemetry_job.go (hunk 2 above)
Comment (coury-clark) [reply enterprise/cmd/worker/internal/telemetry/telemetry_job.go:325]: Ah yeah, thanks. This was just an oversight, I meant to check for the error.

**Pull request merged**